backend/cmm_helpers.mli

```diff
@@ -325,24 +325,25 @@ val make_float_alloc :
   expression list ->
   expression
 
-module Flat_suffix_element : sig
-  type t =
-    | Tagged_immediate
-    | Naked_float
-    | Naked_float32
-    | Naked_int32
-    | Naked_int64_or_nativeint
-end
+(** Allocate a closure block. *)
+val make_closure_alloc :
+  mode:Lambda.alloc_mode ->
+  Debuginfo.t ->
+  tag:int ->
+  expression list ->
```

Comment:
There should probably be a comment saying what the expression and memory_chunk parameters actually correspond to, since these presumably cover function slots as well, and we don't normally talk about the memory chunk types of those I don't think?

Reply:
I don't mind a bit more clarification, but I think the meaning is obvious: each expression is stored in the block, so the `memory_chunk` corresponds to which store operation we want to use. It used to be `caml_initialize` everywhere, even for function pointers, but now we can use the proper operation (which is `Word_int` for function pointers, arity fields and infix headers).
Note that it doesn't matter for small blocks (<256 words), since we use `Calloc` in that case, letting `Selection` do the work of computing the correct sizes and machtypes.

Reply:
I'll try and add some documentation.

Originally, I though about having an `(expr * memory_chunk) list` but it would have been more costly in most cases (because most of the time, the list of memory_chunks is not used, and you always need the list of expressions anyway). In fine, the actual problem here is that it's not possible (currently) to have a function that gives the `memory_chunk` (or `machtype`) of a cmm expression, and therefore we have to explicitly thread it everywhere every time we end up needing it. This is an unfortunate consequence of the design of cmm currently.

Reply:
In particular I think the comment should explicitly say that these lists include the function slots, rather than just the environment.

Reply:
done